### PR TITLE
feat: Guarantee IP Address assignment to DUBBD's ingressgateways

### LINKED ISSUE #8: Ability to guarantee which ingressgateway gets which IP address
Currently, I'm only able to set one range of IP addresses for MetalLB to hand out, by setting the `IP_ADDRESS_POOL` variable.

The problem with this is, that there is no way for me to guarantee that a particular loadbalancer service will get a particular address, which makes it much more difficult to deal with figuring out how to set up my DNS properly.

Proposed solution:

- Eliminate the `IP_ADDRESS_POOL` variable
- Introduce 3 new variables: `IP_ADDRESS_ADMIN_INGRESSGATEWAY`, `IP_ADDRESS_KEYCLOAK_INGRESSGATEWAY`, and `IP_ADDRESS_TENANT_INGRESSGATEWAY`
- Create 3 IPAddressPools when deploying this package rather than just one. One named `ippool-admin-ingressgateway`, one named `ippool-keycloak-ingressgateway`, and one named `ippool-tenant-ingressgateway`. All 3 would use the `namespaceSelectors` and `serviceSelectors` functionality to automatically assign the correct pool to the correct ingressgateway.  See [the docs](https://metallb.universe.tf/configuration/_advanced_ipaddresspool_configuration/#reduce-scope-of-address-allocation-to-specific-namespace-and-service) for more info. In all 3 cases the string `/32` would be appended to the IP address provided to make it a CIDR range of exactly 1 address.
- Introduce new variable `DEFAULT_IP_ADDRESS_POOL` that allows the user to assign whatever other pool they want for all other use cases, associated with a 4th IpAddressPool named `ippool-default`. Alternatively, keep the `IP_ADDRESS_POOL` variable and use it for this.
- Add logic so that if any of the 3 new variables is not set then the associated IPAddressPool is not created

Pros:
- I'm able to guarantee with 100% certainty that each of the different ingressgateways get exactly the IP address that I want it to get, so that I am able to pre-configure my DNS rather than have to wait to configure it after deployment (and risk it possibly changing)

Cons:
- This package becomes coupled to the pattern that DUBBD uses for istio ingressgateways. If this package is not being used in concert with DUBBD it may do stuff for no reason.

### PR DESCRIPTION
**Pull request by RothAndrew**
Closes #8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,7 +29,35 @@ $ ip addr
...
+    --set IP_ADDRESS_KEYCLOAK_INGRESSGATEWAY=10.0.0.33 \
+    --set IP_ADDRESS_TENANT_INGRESSGATEWAY=10.0.0.34 \
+    --confirm
+```
+
+> Notes:
+>   - The IP addresses used here are placeholders. You can use whatever values you want that work for your environment.
+>   - Package versions can be found [here](https://github.com/defenseunicorns/uds-package-metallb/pkgs/container/packages%2Fmetallb)
+>   - If you won't be using Keycloak, you can omit setting the `IP_ADDRESS_KEYCLOAK_INGRESSGATEWAY` variable. The package is smart enough to not create the associated IPAddressPool if the variable is not set.
+>   - If you also want a 4th default IPAddressPool you can additionally set the `IP_ADDRES_POOL` variable too. It should be an IP range, not a single address unlike the other variables which are single address. Ranges can be specified in either CIDR notation or "StartAddress-EndAddress" notation.
```
Comment (mjnagel) [inline README.md:51]:
```suggestion
>   - If you also want a 4th default IPAddressPool you can additionally set the `IP_ADDRESS_POOL` variable too. It should be an IP range, not a single address unlike the other variables which are single address. Ranges can be specified in either CIDR notation or "StartAddress-EndAddress" notation.
```
Typo.

File: charts/addresspools/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+# This version should stay in sync with the version in the zarf.yaml file.
+version: 0.0.2
```
Comment (mjnagel) [inline charts/addresspools/Chart.yaml:19]: We'll want to bump this to 0.0.3 in order to release this - I'm okay with doing that in this PR or we could do a follow on one to add release-please.

**Review by mjnagel — COMMENTED** [review]
Would also be nice to update the zarf-config.yaml with empty defaults for each of these new vars - https://github.com/defenseunicorns/uds-package-metallb/blob/feature/8-guarantee-ip-address-assignment/zarf-config.yaml

File: charts/addresspools/Chart.yaml (hunk 2 above)
Comment (RothAndrew) [reply charts/addresspools/Chart.yaml:19]: If you are good with the plan to use release please I can get it set up and have release please bump it

**Comment by RothAndrew** [comment]
Not sure what you mean. Zarf.yaml sets defaults. Zarf-config.yaml has some create-time stuff in it but no deploy time stuff

File: charts/addresspools/Chart.yaml (hunk 2 above)
Comment (mjnagel) [reply charts/addresspools/Chart.yaml:19]: Works for me.

**Comment by mjnagel** [comment]
@RothAndrew we can leave it off - I was basically thinking of adding deploy time `""` defaults just as an example to show what can be set. Would look similar to what DUBBD has - https://github.com/defenseunicorns/uds-package-dubbd/blob/main/defense-unicorns-distro/zarf-config.yaml . I think in some cases its nicer to set multiple vars via a config file vs `--set` when deploying. Having a "sample" zarf-config can be nice.

**Review by mjnagel — APPROVED** [review]

**Pull request merged**